src/main/java/com/google/api/generator/engine/ast/VariableExpr.java

```diff
@@ -133,7 +143,20 @@ public VariableExpr build() {
                   })
               .collect(Collectors.toList()));
 
+      // Remove duplicates while maintaining insertion order.
+      ImmutableList<AnnotationNode> processedAnnotations = annotations();
+      setAnnotations(
+          new LinkedHashSet<>(processedAnnotations).stream().collect(Collectors.toList()));
```

Comment:
The `ImmutableList` can have duplicates. For example if you attempt to add 2 `@Deprecated` to a variable, will result in code that cannot compile if not de-duped.
That said, I don't feel too strongly about this check itself, but this duplicate check at least does no harm. 
I am really trying to follow similar logic in `MethodDefinition` [here](https://github.com/googleapis/gapic-generator-java/blob/66782d1109fae176b0f3266a3dacb4b55cf3b677/src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java#L256-L258). Adding annotation to `VariableExpr` should be parallel to adding annotation to `MethodDefinition` (except `@Override` is only applicable to method, so `isOverride()` is irrelevant here). Thus I would rather avoid having different logics in builder method here vs. `MethodDefinition`.

Reply:
I see, I think it was a poor decision to use a List and de dup it later, but since we already have a pattern, let's follow it for now and refactor it later.